x-pack/solutions/security/plugins/security_solution_serverless/server/plugin.ts

```diff
@@ -84,6 +90,29 @@ export class SecuritySolutionServerlessPlugin
     // Setup project uiSettings whitelisting
     pluginsSetup.serverless.setupProjectSettings(SECURITY_PROJECT_SETTINGS);
 
+    // Serverless Advanced Settings setup
+    coreSetup
+      .getStartServices()
+      .then(async ([_, depsStart]) => {
+        await new Promise<void>(async (resolve) => {
```

Comment:
Why do we need to create a new Promise here? 🤔

Reply:
I thought it was because the setup was finishing before the register got called but i cannot reproduce now. ill get rid of this